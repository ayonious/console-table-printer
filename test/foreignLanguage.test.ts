import { Table } from '../index';

describe('Example: Print a simple Table with cell colors', () => {
  it('cell colors are working', () => {
    // Create a table
    const p = new Table();

    // add rows with color
    p.addRows([
      // adding multiple rows are possible
      {
        Description: 'Some alphabets 这里是中文这里是中文这里是中文',
        'Ticket No': 'ISSUE-1231',
      },
      {
        Description: 'Some Summary 这里是中文这里是中文',
        'Ticket No': 'ISSUE-22222',
      },
      {
        Description: 'Description 这里是中文',
        'Ticket No': 'ISSUE-Foreign',
      },
    ]);

    // print
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });
});
