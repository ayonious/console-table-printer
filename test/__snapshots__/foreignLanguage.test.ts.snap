// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example: Print a simple Table with cell colors coloring is working 1`] = `
"┌─────────────┬───────────────────────────────────────────────┐
│[37m [0m[01m  Ticket No[0m │[37m [0m[01m                                  Description[0m │
├─────────────┼───────────────────────────────────────────────┤
│[37m [0m[34m ISSUE-1231[0m │[37m [0m[34mSome alphabets 这里是中文这里是中文这里是中文[0m │
│[37m [0m[31mISSUE-22222[0m │[37m [0m[31m            Some Summary 这里是中文这里是中文[0m │
│[37m [0m[01m    ISSUE-1[0m │[37m [0m[01m                       Description 这里是中文[0m │
└─────────────┴───────────────────────────────────────────────┘"
`;

exports[`Example: Print a simple Table with cell colors customized border is working 1`] = `
"[31m╔[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m╦[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m╗[0m
[31m║[0m[37m [0m[01m  Ticket No[0m [31m║[0m[37m [0m[01m                                  Description[0m [31m║[0m
[31m╟[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m╬[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m╢[0m
[31m║[0m[37m [0m[34m ISSUE-1231[0m [31m║[0m[37m [0m[34mSome alphabets 这里是中文这里是中文这里是中文[0m [31m║[0m
[31m║[0m[37m [0m[31mISSUE-22222[0m [31m║[0m[37m [0m[31m            Some Summary 这里是中文这里是中文[0m [31m║[0m
[31m║[0m[37m [0m[01m    ISSUE-1[0m [31m║[0m[37m [0m[01m                       Description 这里是中文[0m [31m║[0m
[31m╚[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m╩[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m═[0m[31m╝[0m"
`;

exports[`Example: Print a simple Table with cell colors foreign alphabets are working 1`] = `
"┌───────────────────────────────────────────────┬───────────────┐
│[37m [0m[01m                                  Description[0m │[37m [0m[01m    Ticket No[0m │
├───────────────────────────────────────────────┼───────────────┤
│[37m [0m[37mSome alphabets 这里是中文这里是中文这里是中文[0m │[37m [0m[37m   ISSUE-1231[0m │
│[37m [0m[37m            Some Summary 这里是中文这里是中文[0m │[37m [0m[37m  ISSUE-22222[0m │
│[37m [0m[37m                       Description 这里是中文[0m │[37m [0m[37mISSUE-Foreign[0m │
└───────────────────────────────────────────────┴───────────────┘"
`;
